packages/contracts-bedrock/src/L1/SuperchainConfig.sol

```diff
@@ -3,98 +3,169 @@ pragma solidity 0.8.15;
 
 // Contracts
 import { Initializable } from "@openzeppelin/contracts/proxy/utils/Initializable.sol";
-
-// Libraries
-import { Storage } from "src/libraries/Storage.sol";
+import { ReinitializableBase } from "src/universal/ReinitializableBase.sol";
 
 // Interfaces
 import { ISemver } from "interfaces/universal/ISemver.sol";
 
+// Libraries
+import { LibString } from "@solady/utils/LibString.sol";
+import { Storage } from "src/libraries/Storage.sol";
+
 /// @custom:proxied true
 /// @custom:audit none This contracts is not yet audited.
 /// @title SuperchainConfig
 /// @notice The SuperchainConfig contract is used to manage configuration of global superchain values.
-contract SuperchainConfig is Initializable, ISemver {
+/// @dev WARNING: When upgrading this contract, any active pause states will be lost as the pause state
+///      is stored in storage variables that are not preserved during upgrades. Therefore, this contract
+///      should not be upgraded while the system is paused.
+contract SuperchainConfig is Initializable, ReinitializableBase, ISemver {
+    /// @notice Thrown when a caller is not the guardian but tries to call a guardian-only function
+    error SuperchainConfig_OnlyGuardian();
+
+    /// @notice Thrown when attempting to pause an identifier that is already paused
+    error SuperchainConfig_AlreadyPaused(address identifier);
+
     /// @notice Enum representing different types of updates.
     /// @custom:value GUARDIAN            Represents an update to the guardian.
     enum UpdateType {
         GUARDIAN
     }
 
-    /// @notice Whether or not the Superchain is paused.
-    bytes32 public constant PAUSED_SLOT = bytes32(uint256(keccak256("superchainConfig.paused")) - 1);
+    /// @notice The duration after which a pause expires. This value is set to exactly 3 months in
+    ///         seconds. Any duration longer than this value is incompatible with Stage 1.
+    uint256 internal constant PAUSE_EXPIRY = 7889238;
 
     /// @notice The address of the guardian, which can pause withdrawals from the System.
     ///         It can only be modified by an upgrade.
-    bytes32 public constant GUARDIAN_SLOT = bytes32(uint256(keccak256("superchainConfig.guardian")) - 1);
+    address public guardian;
+
+    /// @notice Mapping of pause identifiers to their pause timestamps
+    mapping(address => uint256) public pauseTimestamps;
 
     /// @notice Emitted when the pause is triggered.
     /// @param identifier A string helping to identify provenance of the pause transaction.
     event Paused(string identifier);
 
     /// @notice Emitted when the pause is lifted.
-    event Unpaused();
+    event Unpaused(string identifier);
```

Comment:
Since we're breaking the interface on this contract already, I think we should just go ahead and change the type:

```suggestion
    event Paused(address identifier);

    /// @notice Emitted when the pause is lifted.
    event Unpaused(address identifier);
```